menu/templatetags/menubuilder.py

```diff
@@ -92,7 +92,15 @@ def get_items(menu_name, current_path, user):
     if not menu:
         return []
 
-    for i in MenuItem.objects.filter(menu=menu).order_by('order'):
+    if (str(user) == 'AnonymousUser' or str(user) == 'None'):
```

Comment:
If you look at line #74, there are some checks for user type - ie anonymous or logged-in. Can you please add to those for is_staff, and build it into the cache key?  Lastly, you can go back to a single `MenuItem.objects()` query and build in the is_staff filtering on line #114 (see `menuitems.append` when conditions are met). 

This keeps your change in line with the previous structure of this file. It also means all similar checks occur in a similar manner.